uaa/src/test/java/org/cloudfoundry/identity/uaa/orchestrator/OrchestratorClaimAT.java

```diff
@@ -1,124 +1,343 @@
+import org.springframework.http.client.HttpComponentsClientHttpRequestFactory;
+import org.springframework.security.oauth2.client.DefaultOAuth2ClientContext;
+import org.springframework.security.oauth2.client.OAuth2RestOperations;
+import org.springframework.security.oauth2.client.OAuth2RestTemplate;
+import org.springframework.security.oauth2.client.resource.OAuth2ProtectedResourceDetails;
+import org.springframework.security.oauth2.client.token.DefaultAccessTokenRequest;
+import org.springframework.security.oauth2.client.token.grant.client.ClientCredentialsResourceDetails;
 import org.springframework.test.context.ContextConfiguration;
 import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
 
-import com.ge.predix.iam.kubernetes.enums.ServiceName;
-import com.ge.predix.iam.kubernetes.exception.ServiceInstanceProviderException;
-import com.ge.predix.iam.kubernetes.model.Claim;
-import com.ge.predix.iam.kubernetes.utils.AssumeRoleUtil;
-import com.ge.predix.iam.kubernetes.utils.OrchestratorUtil;
+import java.util.Random;
+
+
+import static org.junit.Assert.*;
 
 @RunWith(SpringJUnit4ClassRunner.class)
 @ContextConfiguration(classes = OrchestratorIntegrationTestConfig.class)
+@Slf4j
 public class OrchestratorClaimAT {
+    protected static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    private static final Logger LOGGER = LoggerFactory.getLogger(OrchestratorClaimAT.class);
 
-    @Value("${kubernetes.config.secondary-namespace}")
-    private String namespace;
+    public static final String ZONES_URI = "/orchestrator/zones";
+
+    private UaaRequest uaaRequest;
+
+    protected static HttpHeaders headers;
+    private Random random = new Random();
+
+    private  final String UAA_SUBDOMAIN = "rest-uaa-subdomain"+random.nextInt(1000);
+
+    private static final String UAA_ADMIN_CLIENT_SECRET = "ZHVtbXlBZG1pblNlY3JldDEyMyMh";
+
+    private static final String UAA_CLAIM_NAME_PREFIX = "uaa-it-";
+    @Value("${uaa.url}")
+    private String uaaUrl;
+
+    @Value("${uaa.orchestrator.clientId}")
+    private String clientId;
+
+    @Value("${uaa.orchestrator.clientSecret}")
+    private String clientSecret;
+
+    @Value("${kubernetes.config.env}")
+    private String environment;
 
     @Autowired
     private OrchestratorUtil orchestratorUtil;
 
-    @Autowired
-    private AssumeRoleUtil assumeRoleUtil;
+    protected static OAuth2RestTemplate trustedIssuerValidScopesRestTemplate;
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(OrchestratorClaimAT.class);
-    private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+    @Value("${kubernetes.config.secondary-namespace:iam-qasvc}")
+    private String namespace;
 
-    @BeforeClass
-    public static void setupClassLevelData() {
-        LOGGER.info("In OrchestratorClaimAT.setupClassLevelData()");
-    }
+
+    private Integer uaaInstanceCount = random.nextInt(1000);
+
+    private String claimName;
```

Comment:
No need for this variable to be global. It's only used in testCreateZone(). Declare it there

Reply:
Done

Reply:
Done